### PR TITLE
Unit tests for incrementing a keyword

### PR DESCRIPTION
**Pull request by kirinrastogi**
I added unit tests, and did some small changes.

Next I am refactoring the incrementation process to be handled on the database.

New keywords can no longer be made through voting.

### CODE REVIEW

File: routes/keyword.js [hunk 1]
```diff
@@ -51,15 +51,12 @@ router.get('/:word', (req, res) => {
...
+  return new Promise((resolve, reject) => {
     Keyword.findOne({ word })
       .then((keyword) => {
         let oldSum = 0
         if (!keyword) {
-          keyword = new Keyword({
-            word,
-            votes: [],
-          })
+          reject(new Error('Word not found in database'))
```
Comment (JackyChiu) [inline routes/keyword.js]: What happens on fail now? Does the keyword just not get incremented?

File: test/keyword.test.js [hunk 2]
```diff
@@ -34,6 +34,68 @@ function checkValidError(res) {
...
 
+function checkKeywordAfterIncrement(first, last, vote) {
+  first._id.should.equal(last.id)
+  const len = first.votes.length
+  first.votes[len - 1].sum.should.equal(last.oldSum)
+  first.votes[len - 1].sum.should.equal(last.newSum - vote)
+}
+
+function getKeywordData(word) {
+  return new Promise((resolve) => {
```
Comment (JackyChiu) [inline test/keyword.test.js]: We might as well add the reject since your catching in the blocks below

File: routes/keyword.js (hunk 1 above)
Comment (kirinrastogi) [inline routes/keyword.js]: Correct, the keyword can only be voted on if it exists. It should be created when the article is added, if the keyword does not exist.

Comment (JackyChiu) [inline routes/keyword.js]: @kirinrastogi Is issue #62 related to this change?

Comment (kirinrastogi) [inline routes/keyword.js]: 1000% yes because we never readded keyword creation, I have another branch that Im working on it in

Comment (JackyChiu) [inline routes/keyword.js]: @kirinrastogi HOT FIX INCOMING 💯

**Pull request merged**